applications/adv_networking_bench/cpp/CMakeLists.txt

```
@@ -104,43 +104,54 @@ install(
 
 # Add testing
 if(BUILD_TESTING)
-
-  ## adv_networking_bench_default_tx_rx_test ##
-  # The test below is expected to run on an IGX devkit with a loopback
-  # connecting both ports of the IGX NIC (default config currently hardcodes
-  # the BUS address of these two ports), with optimal system setups (BAR1,
+  ## adv_networking_bench_dpdk_loopback_test ##
+  # The test below is expected to run on a system with a ConnectX NIC with
+  # both ports connected in a loopback, with optimal system setups (BAR1,
   # GPU clock, CPU clock, huge pages, etc... docs incoming).
 
-  # Get the BUS ID of the interface doing Rx in the config
-  execute_process(COMMAND bash "-c" "grep -A2 -B2 ' rx:' ${CMAKE_CURRENT_SOURCE_DIR}/../adv_networking_bench_default_tx_rx.yaml | grep ' address:' | awk '{print $2}'"
-                  OUTPUT_VARIABLE rx_bus
-                  COMMAND_ERROR_IS_FATAL ANY
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  message(DEBUG "[ANO Test] Rx interface BUS info from config: ${rx_bus}")
-
-  # Get the interface name
-  execute_process(COMMAND bash "-c" "find /sys/class/net -exec bash -c 'echo {} $(readlink -f {}/device)' \; | grep ${rx_bus} | awk '{print $1}'"
-                  OUTPUT_VARIABLE rx_if
-                  COMMAND_ERROR_IS_FATAL ANY
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  message(DEBUG "[ANO Test] Interface for ${rx_bus}: ${rx_if}")
-
-  # Get the ETH mac address
-  execute_process(COMMAND cat "${rx_if}/address"
-                  OUTPUT_VARIABLE rx_eth_mac
-                  COMMAND_ERROR_IS_FATAL ANY
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  message(DEBUG "[ANO Test] MAC address for interface ${rx_if}: ${rx_eth_mac}")
-
-  # Inject in test config + add app timeout
+  # Define a macro to handle the execution and checks
+  macro(execute_and_check MSG CMD OUTPUT_VAR)
+    execute_process(COMMAND bash "-c" "${CMD}"
+                    RESULT_VARIABLE result_code
+                    OUTPUT_VARIABLE ${OUTPUT_VAR}
+                    ERROR_VARIABLE error_code
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(NOT result_code EQUAL 0 OR "${${OUTPUT_VAR}}" STREQUAL "")
+      message(WARNING "[ANO Test] ${MSG} '${CMD}' failed (${result_code}) or output was empty")
+      message(STATUS "[ANO Test] Skipping advanced networking loopback test")
+      return()
+    else()
+      message(STATUS "[ANO Test] ${MSG} ${${OUTPUT_VAR}}")
+    endif()
+  endmacro()
+
+  # Check NIC addresses
+  message(STATUS "[ANO Test] Checking system compatibility for advanced networking loopback test")
+  execute_and_check("ConnectX BUS Addresses:" "lspci | grep -E \"Ethernet.*ConnectX\" | awk '{print $1}'" cx_bus_address_list)
```

Per @cliffburdick , will update to

```suggestion
  execute_and_check("NVIDIA NIC BUS Addresses:" "lspci -nn | grep -iE 'Ethernet controller.*\[15b3:' | awk '{print $1}'" cx_bus_address_list)
```